tests/monster_test.fbs

```diff
@@ -67,6 +67,9 @@ table Monster {
   testf:float = 3.14159 (id:25);
   testf2:float = 3 (id:26);
   testf3:float (id:27);
+  test_hex1:int = -0x00 (id:29);
```

Comment:
This is going to add fields in the generated code for a lot of languages (which are not in your commit).

Much simpler to add a case to `ValueTest()` in `test.cpp`